system-tests/lib/performance-reporter.js

```diff
@@ -1,36 +1,110 @@
 const path = require('path')
 const chalk = require('chalk')
 const Libhoney = require('libhoney')
+const { v4: uuidv4 } = require('uuid')
 
-const pkg = require('@packages/root')
 const ciProvider = require('@packages/server/lib/util/ci_provider')
 const { commitInfo } = require('@cypress/commit-info')
+const { getNextVersionForPath } = require('../../scripts/get-next-version')
 
-class StatsdReporter {
+const honey = new Libhoney({
+  dataset: 'systemtest-performance',
+  writeKey: process.env.HONEYCOMB_API_KEY,
+})
+
+const spanId = process.env.CIRCLE_WORKFLOW_ID || uuidv4()
+const circleCiRootEvent = honey.newEvent()
+
+// Mocha events ('test', 'test end', etc) have no way to wait
+// for async callbacks, so we can't guarantee we have this
+// data ready by the time any of the reporter's events are emitted.
+
+// Therefore, we have each honeycomb event await this promise
+// before sending itself.
+let asyncInfo = Promise.all([getNextVersionForPath('../../packages'), commitInfo()])
+.then(([nextVersion, commitInformation]) => {
+  const ciInformation = ciProvider.commitParams() || {}
+
+  return {
+    nextVersion,
+    branch: commitInformation.branch || ciInformation.branch,
+    commitSha: commitInformation.sha || ciInformation.sha,
+  }
+})
+
+function addAsyncInfoAndSend (honeycombEvent) {
+  return asyncInfo.then((info) => {
+    honeycombEvent.add(info)
+    honeycombEvent.send()
+  })
+}
+
+circleCiRootEvent.timestamp = Date.now()
+circleCiRootEvent.add({
+  buildUrl: process.env.CIRCLE_BUILD_URL,
+  platform: process.platform,
+  arch: process.arch,
+
+  spanId,
+  traceId: spanId,
+})
+
+// This file is executed once as a script at the beginning of the circleci build,
+// so that we can send the root event exactly once and associate all the various
+// system test tasks and build steps into a single span.
+if (require.main === module) {
```

Comment:
nit: IMO it would be easier for new people coming in to debug this if this "script" was in `system-tests/scripts` and not just conditionally running if this `module` is `main`. Separation of the script and the lib is how most of our scripts are set up.

Reply:
Done.